R/connectAndRunGWASAnalysis.R

```diff
@@ -0,0 +1,196 @@
+#' @param controls_description PARAM_DESCRIPTION, Default: cohorts_settings$controls_cohort$name
+#' @param notification_email PARAM_DESCRIPTION, Default: connection_sandboxAPI$notification_email
+#' @param release PARAM_DESCRIPTION, Default: 'Regenie9'
+#' @export
+#' @importFrom stringr str_detect
+#' @importFrom dplyr bind_rows
+#' @importFrom dplyr %>%
+#' @importFrom tibble tibble
+#' @importFrom readr write_tsv
+#' @importFrom httr add_headers upload_file POST status_code content
+#' @importFrom jsonlite toJSON
+runGWASAnalysis <- function(
+    connection_sandboxAPI,
+    cohorts_settings,
+    phenotype_name,
+    title = phenotype_name,
+    description = phenotype_name,
+    cases_description = cohorts_settings$cases_cohort$name,
+    controls_description = cohorts_settings$controls_cohort$name,
+    notification_email = connection_sandboxAPI$notification_email,
+    release = "finngen_R12"
+) {
+
+  if(!stringr::str_detect(phenotype_name, "^[[:upper:]|[:digit:]]+$")){
+    stop("phenotype_name must contain only in capital letters or numbers" )
+  }
+
+  # create phenofile
+  tmp_path_phenofile = file.path(tempdir(), "phenofile.tsv")
+
+  dplyr::bind_rows(
+    tibble::tibble( FID = cohorts_settings$cases_cohort$validated_ids, {{phenotype_name}}:=1),
+    tibble::tibble( FID = cohorts_settings$controls_cohort$validated_ids, {{phenotype_name}}:=0)
+  ) %>% readr::write_tsv(tmp_path_phenofile)
+
+  # prepare api params
+  authorization = paste("Bearer", connection_sandboxAPI$token)
+  headers = httr::add_headers(c('Authorization'=authorization, 'Content-Type'="multipart/form-data"))
+
+  url = paste0(connection_sandboxAPI$base_url, "v2/gwas")
+
+  json = jsonlite::toJSON(
+    list(
+      num_cases=length(cohorts_settings$cases_cohort$validated_ids),
+      num_controls=length(cohorts_settings$controls_cohort$validated_ids),
+      title = title,
+      description = description,
+      phenotype_name = phenotype_name,
+      cases = cases_description,
+      controls = controls_description,
+      notification_email = notification_email,
+      release = release
+    ),
+    auto_unbox = TRUE)
+
+  body = list(data=json, phenofile=httr::upload_file(tmp_path_phenofile))
+
+  logTibble = connection_sandboxAPI$conn_status_tibble$logTibble
+
+  if(logTibble$type == "ERROR"){
+    res <- list(
+      status = FALSE,
+      message = "Connection in connection_sandboxAPI not stablised"
+    )
+  }else{
+    # post call
+    res <- tryCatch({
+      res <- httr::POST(url, body=body, headers)
+      res <- list(
+        status = httr::status_code(res) == 200,
+        content = paste(httr::content(res), collapse = "\n")
+      )
+    },
+    error = function(cond) {
+      return(list(
+        status = FALSE,
+        message = stringr::str_c("Un expected error in runGWASAnalysis", cond$message)
+      ))
+    })
+  }
+
+
+  return(res)
+
+}
+
+
+
+.get_cohorts_settings <- function(
```

Comment:
It is not needed in CO2 after all - I thought it might be, so I left it here. Only if it can be useful for the users - with this function you can prepare an input structure from the exported CSV cohort file.  Probably it is quite straightforward to do, so no need in it. What do you think?